bycycle/group/features.py

```diff
@@ -189,81 +269,90 @@ def compute_features_3d(sigs, fs, f_range, compute_features_kwargs=None,
     ...                                compute_features_kwargs=compute_kwargs)
     """
 
+    n_jobs = cpu_count() if n_jobs == -1 else n_jobs
+    df_features = []
+
     # Convert list of kwargs to array to check dimensions
-    kwargs = compute_features_kwargs
+    kwargs = deepcopy(compute_features_kwargs)
+    kwargs = np.array(kwargs) if isinstance(kwargs, list) else kwargs
 
-    if isinstance(kwargs, list):
-        kwargs = np.array(kwargs)
-    elif kwargs is None:
-        kwargs = {}
+    check_kwargs_shape(sigs, kwargs, axis)
+    kwargs = list(kwargs.flatten()) if isinstance(kwargs, np.ndarray) else [kwargs]
 
-    # Ensure compute_features corresponds to sigs
-    if isinstance(kwargs, np.ndarray):
+    # Compute features
+    df_features = np.zeros((np.shape(sigs)[0], np.shape(sigs)[1])).tolist()
 
-        if kwargs.ndim == 1 and np.shape(kwargs)[0] != np.shape(sigs)[0]:
+    if axis == 0:
```

Comment:
The process for axis 0 or 1 is effectively the same, with just a bit of data manipulation depending which actual axis. 

This can therefore potentially be condensed, I think:

```
if axis in [0, 1]:
    sigs = np.swapaxes(sigs, 0, 1) is axis == 1 else sigs
    # then have the lines 287 to 295, that are otherwise duplicated in 300 to 308
    if axis == 1: df_features = [list(dfs) for dfs in zip(*df_features)]
```

Reply:
Good idea! Updated to this